packages/edit-navigation/src/components/inspector-additions/manage-locations.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useContext } from '@wordpress/element';
+import {
+	Spinner,
+	SelectControl,
+	CheckboxControl,
+	Button,
+	Modal,
+	PanelBody,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useMenuLocations, MenuIdContext } from '../../hooks';
+
+export default function ManageLocations( {
+	onSelectMenu,
+	isModalOpen,
+	openModal,
+	closeModal,
+} ) {
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus(), [] );
+
+	const selectedMenuId = useContext( MenuIdContext );
+
+	const {
+		menuLocations,
+		assignMenuToLocation,
+		toggleMenuToLocation,
+	} = useMenuLocations();
+	const menusWithSelection = menuLocations.map( ( themeLocation ) => (
+		<CheckboxControl
+			key={ themeLocation.name }
+			checked={ themeLocation.menu === selectedMenuId }
+			onChange={ () =>
+				toggleMenuToLocation( themeLocation.name, selectedMenuId )
+			}
+			label={ themeLocation.name }
+		/>
+	) );
+	if ( ! menus || ! menuLocations ) {
+		return <Spinner />;
+	}
+
+	if ( ! menus.length ) {
+		return <p>{ __( 'There are no available menus.' ) }</p>;
+	}
+
+	if ( ! menuLocations.length ) {
+		return <p>{ __( 'There are no available menu locations.' ) }</p>;
+	}
+
+	const menuLocationCard = menuLocations.map( ( menuLocation ) => (
+		<div
+			key={ menuLocation.name }
+			className="edit-navigation-header__manage-locations__menage-locations-modal__menu-entry"
```

Comment:
These classnames are very long and break with the standard, which is ``${ packageName }${ rootComponentName }__${ semanticNameForElement }`` (https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/coding-guidelines.md#css).

To be completely true to the guidelines, it could be `edit-navigation-inspector-additions__manage-locations-menu-entry` which is still pretty long. I think you could also consider breaking the rules slightly and making it `edit-navigation-manage-locations__menu-entry`. I don't think this 'root component' idea has to always be adhered to for subcomponents, as long as we're sure there won't be two ManageLocations components, then it will be fine.

Also seeing the typo 'menage' across the PR.